src/Microsoft.Azure.Relay/RelayEventSource.cs

```diff
@@ -391,11 +568,54 @@ static string DateTimeToString(DateTime dateTime)
             return dateTime.ToString(CultureInfo.InvariantCulture);
         }
 
+        static TraceDetails PrepareTrace(object source)
+        {
+            string sourceString;
+            TrackingContext trackingContext;
+
+            var traceSource = source as ITraceSource;
+            if (traceSource != null)
+            {
+                trackingContext = traceSource.TrackingContext;
+                SetCurrentThreadActivityId(trackingContext);
+                sourceString = traceSource.ToString();
+            }
+            else
+            {
+                sourceString = CreateSourceString(source);
+                trackingContext = null;
+            }
+
+            return new TraceDetails(sourceString, trackingContext);
+        }
+
+        static void SetCurrentThreadActivityId(TrackingContext trackingContext)
+        {
+#if NET45
```

Comment:
@cesarme I had to add this for .NET Core. I think in later releases this API is supported.